src/main/java/seedu/address/model/medicalnote/MedicalNote.java

```diff
@@ -0,0 +1,102 @@
+package seedu.address.model.medicalnote;
+
+import static seedu.address.commons.util.AppUtil.checkArgument;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Objects;
+
+/**
+ * Represents a medical note entry for a Patient,
+ * on a particular date by a particular doctor.
+ * Guarantees: details are present and not null, date and doctor's names are validated, fields are immutable.
+ */
+public class MedicalNote {
+    public static final String DATE_MESSAGE_CONSTRAINTS = "Date should be of the format \"DD-MM-YYYY\""
+            + "where D, M and Y represent digits of the day, month and year of the date respectively.";
+    public static final String NAME_MESSAGE_CONSTRAINTS =
+            "Names should only contain alphanumeric characters and spaces, and it should not be blank";
+    public static final String CONTENT_MESSAGE_CONSTRAINTS = "Content should not be blank";
+    private static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ofPattern("dd-MM-yyyy");
+    private static final String NAME_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
+    /** Date of medical note entry. */
+    public final LocalDate date;
+    /** Name of doctor entering the note. */
+    public final String doctorName;
+    /** Content of the note i.e. relevant information from patient's consultation. */
+    public final String content;
```

Comment:
Perhaps it is better to have these fields as private and provide getters for encapsulation purposes.

Reply:
Maybe we can refactor this if needed after looking at the implementation? Since this is based on conventions in the other Patient detail classes i.e. Name, Tag, etc.

Reply:
ok we can kiv